src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectionPool/HttpConnectionPool.cs

```diff
@@ -881,11 +885,11 @@ public void Dispose()
                     _availableHttp2Connections.Clear();
                 }
 
-                if (_http3Connection is not null)
+                if (IsHttp3Supported() && _availableHttp3Connections is not null)
                 {
-                    toDispose ??= new();
-                    toDispose.Add(_http3Connection);
-                    _http3Connection = null;
+                    toDispose = [.. _availableHttp3Connections];
```

Comment:
```suggestion
                    toDispose ??= new();
                    toDispose.AddRange(_availableHttp3Connections);
```

`[.. _availableHttp3Connections]` would ignore any http2 connections already in the list.